### PR TITLE
[MOB-4431] Clear lastpushpayload if app is not launched via a notification click

### PR DESCRIPTION
**Pull request by amanjayshahomnicg**

### CODE REVIEW

**Comment by roninopf** [comment]
Hi @amanjayshahomnicg! Can we change this to make `lastPushPayload` just an in memory variable? That way it will match the Android SDK's behavior. I'm also worried that the failing integration tests point to a deeper issue I don't understand.

**Comment by amanforindia** [comment]
Hi @roninopf, the reason I hadn't done it that way is to preserve this advertised feature: https://support.iterable.com/hc/en-us/articles/360035018152-Iterable-s-iOS-SDK-#upgrading-to-6-4-8

**Comment by amanjayshahomnicg** [comment]
Hi @roninopf, the reason I hadn't done it that way is to preserve this advertised feature: https://support.iterable.com/hc/en-us/articles/360035018152-Iterable-s-iOS-SDK-#upgrading-to-6-4-8

**Comment by roninopf** [comment]
> Hi @roninopf, the reason I hadn't done it that way is to preserve this advertised feature: https://support.iterable.com/hc/en-us/articles/360035018152-Iterable-s-iOS-SDK-#upgrading-to-6-4-8

Ah, okay. Please feel free to continue with the change mentioned earlier; I'll handle making sure the documentation is updated to reflect the change on our SDK release.

> Hi @roninopf I reviewed the logs for the E2E test failures and I suspect it could be because the PR is opened from my fork -> the main repo. This might cause the secrets to not be available to this PR. For reference: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks

Ah, that's a good catch. Let me see what I can do to get this working for you. Thanks for linking.

And thank you for your work!

File: swift-sdk/Internal/InternalIterableAPI.swift [hunk 1]
```diff
@@ -519,15 +528,15 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
...
-                                               value: Const.UserDefault.payloadExpiration,
-                                               to: dateProvider.currentDate)
-        localStorage.saveLastPushPayload(payload, withExpiration: expiration)
         
         if let metadata = IterablePushNotificationMetadata.metadata(fromLaunchOptions: payload) {
             if let templateId = metadata.templateId {
                 attributionInfo = IterableAttributionInfo(campaignId: metadata.campaignId, templateId: templateId, messageId: metadata.messageId)
             }
+
+            if !metadata.isGhostPush {
```
Comment (harrymash2006) [inline swift-sdk/Internal/InternalIterableAPI.swift:537]: save lastpuspayload only if it is NOT a ghostpush (as per Android)

Comment (devcsomnicg) [inline swift-sdk/Internal/InternalIterableAPI.swift:537]: save lastpuspayload only if it is NOT a ghostpush (as per Android)

**Comment by devcsomnicg** [comment]
@roninopf Could you start reviewing this PR ?

**Comment by roninopf** [comment]
> 

Will do and get started on! Thanks!

**Review by roninopf — APPROVED** [review]

**Comment by roninopf** [comment]
@devcsomnicg The changes look good; if you'd like to try to fix some of the unit tests (not the integration tests; these can be run with command + U), please feel free, but otherwise I can do it as they are small changes and I'm already familiar with the unit testing structure.

**Comment by devcsomnicg** [comment]
> @devcsomnicg The changes look good; if you'd like to try to fix some of the unit tests (not the integration tests; these can be run with command + U), please feel free, but otherwise I can do it as they are small changes and I'm already familiar with the unit testing structure.

ok, we will work on unit tests tomorrow

**Comment by roninopf** [comment]
@devcsomnicg Looks like there's one last test that just needs to be deleted (`NotificationResponsesTests.testSavePushPayload()`), but it's otherwise done!

**Comment by roninopf** [comment]
Merging in. Thanks for your work on this!

**Pull request merged**